src/dynamic_boundary_conditions/hyetograph.py

```diff
@@ -1,63 +1,326 @@
 # -*- coding: utf-8 -*-
 """
 @Script name: hyetograph.py
-@Description:
+@Description: Create interactive hyetograph plots for sites within the catchment area.
 @Author: pkh35
-@Date: 17/01/2022
 @Last modified by: sli229
-@Last modified date: 21/10/2022
+@Last modified date: 9/12/2022
 """
 
-import geopandas as gpd
+import logging
 import pathlib
-from shapely.geometry import Polygon
+import pandas as pd
+import numpy as np
+from typing import Literal
+from math import floor, ceil
+from scipy.interpolate import interp1d
+import plotly.express as px
 from src.digitaltwin import setup_environment
-from src.dynamic_boundary_conditions import rainfall_sites
-from src.dynamic_boundary_conditions import thiessen_polygon_calculator
-from src.dynamic_boundary_conditions import hirds_rainfall_data_to_db
-from src.dynamic_boundary_conditions import hirds_rainfall_data_from_db
+from src.dynamic_boundary_conditions import main_rainfall, thiessen_polygons, hirds_rainfall_data_from_db
 
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
 
-def catchment_area_geometry_info(catchment_file_path) -> Polygon:
+formatter = logging.Formatter("%(levelname)s:%(asctime)s:%(name)s:%(message)s")
+stream_handler = logging.StreamHandler()
+stream_handler.setFormatter(formatter)
+
+log.addHandler(stream_handler)
+
+
+def get_transposed_data(rain_depth_in_catchment: pd.DataFrame) -> pd.DataFrame:
     """
-    Extract shapely geometry polygon from the catchment file.
+    Clean and transpose the retrieved scenario data from the database for sites within the catchment area and
+    return in Pandas DataFrame format.
 
     Parameters
     ----------
-    catchment_file_path
-        The file path of the catchment polygon shapefile.
+    rain_depth_in_catchment : pd.DataFrame
+        Rainfall depths for sites within the catchment area for a specified scenario retrieved from the database.
     """
-    catchment = gpd.read_file(catchment_file_path)
-    catchment = catchment.to_crs(4326)
-    catchment_polygon = catchment["geometry"][0]
-    return catchment_polygon
+
+    # drop unnecessary columns
+    catchment_data = rain_depth_in_catchment.drop(columns=["category", "rcp", "time_period", "ari", "aep"])
+    # change column names
+    for index, column_name in enumerate(catchment_data.columns):
+        if column_name.endswith("m"):
+            catchment_data.columns.values[index] = int(column_name[:-1])
+        elif column_name.endswith("h"):
+            catchment_data.columns.values[index] = int(column_name[:-1]) * 60
```

Comment:
Could you make the comment explain this in a bit more detail

Reply:
extra comments added.